src/internal/feature/feature_test.go

```diff
@@ -0,0 +1,752 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package feature provides feature flags.
+// NOTE: These functions rely on shared global state and therefore cannot be run in parallel.
+package feature
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupFeatureState(t *testing.T, defaults []Feature, user []Feature) {
+	t.Helper()
+
+	// Cleanup first
+	initFeatureState(t)
+
+	// Set features if given
+	if len(defaults) > 0 {
+		defaultFeatures.Store(featuresToMap(defaults))
+	}
+	if len(user) > 0 {
+		userFeatures.Store(featuresToMap(user))
+	}
+}
+
+func initFeatureState(t *testing.T) {
+	t.Helper()
+	defaultFeatures.Store(map[Name]Feature{})
+	userFeatures.Store(map[Name]Feature{})
+}
+
+func TestIsEnabled(t *testing.T) {
+	tt := []struct {
+		name     string
+		user     []Feature
+		defaults []Feature
+		fName    Name
+		expect   bool
+	}{
+		{
+			name:   "neither user nor default enabled",
+			expect: false,
+		},
+		{
+			name: "user enabled, default empty",
+			user: []Feature{
+				{Name: "foo", Enabled: true},
+			},
+			fName:  "foo",
+			expect: true,
+		},
+		{
+			name: " enabled, user empty",
```

Comment:
nit: A few test names are missing the word default

Reply:
Ah dang i thought i fixed most of these 😆